diff --git a/helix-core/src/main/java/org/apache/helix/messaging/AsyncCallback.java b/helix-core/src/main/java/org/apache/helix/messaging/AsyncCallback.java
index e1cdf046fd..b1a9f8c405 100644
--- a/helix-core/src/main/java/org/apache/helix/messaging/AsyncCallback.java
+++ b/helix-core/src/main/java/org/apache/helix/messaging/AsyncCallback.java
@@ -108,7 +108,7 @@ final void setMessagesSent(List<Message> generatedMessage) {
   final synchronized void startTimer() {
     if (_timer == null && _timeout > 0) {
       if (_startTimeStamp == 0) {
-        _startTimeStamp = new Date().getTime();
+        _startTimeStamp = System.currentTimeMillis();
       }
       _timer = new Timer(true);
       _timer.schedule(new TimeoutTask(this), _timeout);
diff --git a/helix-core/src/main/java/org/apache/helix/messaging/handling/HelixStateTransitionHandler.java b/helix-core/src/main/java/org/apache/helix/messaging/handling/HelixStateTransitionHandler.java
index 36245de989..3b4fa0100b 100644
--- a/helix-core/src/main/java/org/apache/helix/messaging/handling/HelixStateTransitionHandler.java
+++ b/helix-core/src/main/java/org/apache/helix/messaging/handling/HelixStateTransitionHandler.java
@@ -337,7 +337,7 @@ public HelixTaskResult handleMessage() {
 
       _statusUpdateUtil.logInfo(message, HelixStateTransitionHandler.class,
           "Message handling task begin execute", manager);
-      message.setExecuteStartTimeStamp(new Date().getTime());
+      message.setExecuteStartTimeStamp(System.currentTimeMillis());
 
       try {
         preHandleMessage();
