diff --git a/query/src/main/java/org/apache/accumulo/examples/wikisearch/iterator/BooleanLogicIterator.java b/query/src/main/java/org/apache/accumulo/examples/wikisearch/iterator/BooleanLogicIterator.java
index d9f3c94..1324006 100644
--- a/query/src/main/java/org/apache/accumulo/examples/wikisearch/iterator/BooleanLogicIterator.java
+++ b/query/src/main/java/org/apache/accumulo/examples/wikisearch/iterator/BooleanLogicIterator.java
@@ -680,7 +680,7 @@ public BooleanLogicTreeNode transformTreeNode(TreeNode node) throws ParseExcepti
             continue;
           }
           String fValue = t.getValue().toString();
-          fValue = fValue.replaceAll("'", "");
+          fValue = fValue.replace("'", "");
           boolean negated = t.getOperator().equals("!=");
           
           if (!fName.startsWith(FIELD_NAME_PREFIX)) {
