diff --git a/tools/perf/src/main/java/org/apache/bookkeeper/tools/perf/table/PerfClient.java b/tools/perf/src/main/java/org/apache/bookkeeper/tools/perf/table/PerfClient.java
index 4b3442700cf..835f816c655 100644
--- a/tools/perf/src/main/java/org/apache/bookkeeper/tools/perf/table/PerfClient.java
+++ b/tools/perf/src/main/java/org/apache/bookkeeper/tools/perf/table/PerfClient.java
@@ -324,7 +324,7 @@ private void runBenchmarkTasks() throws Exception {
 
                     final CountDownLatch latch = new CountDownLatch(tasks.size());
                     @Cleanup("shutdown")
-                    ExecutorService executor = Executors.newCachedThreadPool();
+                    ExecutorService executor = Executors.newFixedThreadPool(tasks.size());
                     for (BenchmarkTask task : tasks) {
                         executor.submit(() -> {
                             try {
