diff --git a/src/main/java/org/apache/joshua/decoder/ff/FeatureVector.java b/src/main/java/org/apache/joshua/decoder/ff/FeatureVector.java
index 34f19ae4..89d729af 100644
--- a/src/main/java/org/apache/joshua/decoder/ff/FeatureVector.java
+++ b/src/main/java/org/apache/joshua/decoder/ff/FeatureVector.java
@@ -338,7 +338,7 @@ public String mosesString() {
 
     // First print all the dense feature names in order
     for (int i = 0; i < DENSE_FEATURE_NAMES.size(); i++) {
-      outputString.append(String.format("%s=%.3f ", DENSE_FEATURE_NAMES.get(i).replaceAll("_", "-"), getDense(i)));
+      outputString.append(String.format("%s=%.3f ", DENSE_FEATURE_NAMES.get(i).replace("_", "-"), getDense(i)));
       printed_keys.add(DENSE_FEATURE_NAMES.get(i));
     }
 
diff --git a/src/main/java/org/apache/joshua/mira/MIRACore.java b/src/main/java/org/apache/joshua/mira/MIRACore.java
index 9ba046dd..3272115e 100755
--- a/src/main/java/org/apache/joshua/mira/MIRACore.java
+++ b/src/main/java/org/apache/joshua/mira/MIRACore.java
@@ -2795,11 +2795,11 @@ private String normalize(String str, int normMethod) {
       return str;
 
     // replace HTML/SGML
-    str = str.replaceAll("&quot;", "\"");
-    str = str.replaceAll("&amp;", "&");
-    str = str.replaceAll("&lt;", "<");
-    str = str.replaceAll("&gt;", ">");
-    str = str.replaceAll("&apos;", "'");
+    str = str.replace("&quot;", "\"");
+    str = str.replace("&amp;", "&");
+    str = str.replace("&lt;", "<");
+    str = str.replace("&gt;", ">");
+    str = str.replace("&apos;", "'");
 
     // split on these characters:
     // ! " # $ % & ( ) * + / : ; < = > ? @ [ \ ] ^ _ ` { | } ~
diff --git a/src/main/java/org/apache/joshua/util/FormatUtils.java b/src/main/java/org/apache/joshua/util/FormatUtils.java
index ead2db70..61fde977 100644
--- a/src/main/java/org/apache/joshua/util/FormatUtils.java
+++ b/src/main/java/org/apache/joshua/util/FormatUtils.java
@@ -128,9 +128,9 @@ public static String escapeSpecialSymbols(String s) {
   }
   
   public static String unescapeSpecialSymbols(String s) {
-    return s.replaceAll("-lsb-", "[")
-            .replaceAll("-rsb-", "]")
-            .replaceAll("-pipe-", "|");
+    return s.replace("-lsb-", "[")
+            .replace("-rsb-", "]")
+            .replace("-pipe-", "|");
   }
   
   /**
diff --git a/src/main/java/org/apache/joshua/zmert/MertCore.java b/src/main/java/org/apache/joshua/zmert/MertCore.java
index 17f66779..47cede9c 100644
--- a/src/main/java/org/apache/joshua/zmert/MertCore.java
+++ b/src/main/java/org/apache/joshua/zmert/MertCore.java
@@ -2597,11 +2597,11 @@ private String normalize(String str, int normMethod) {
     if (normMethod == 0) return str;
 
     // replace HTML/SGML
-    str = str.replaceAll("&quot;", "\"");
-    str = str.replaceAll("&amp;", "&");
-    str = str.replaceAll("&lt;", "<");
-    str = str.replaceAll("&gt;", ">");
-    str = str.replaceAll("&apos;", "'");
+    str = str.replace("&quot;", "\"");
+    str = str.replace("&amp;", "&");
+    str = str.replace("&lt;", "<");
+    str = str.replace("&gt;", ">");
+    str = str.replace("&apos;", "'");
 
 
 
