diff --git a/maven/core/src/main/java/org/apache/karaf/maven/command/MavenConfigurationSupport.java b/maven/core/src/main/java/org/apache/karaf/maven/command/MavenConfigurationSupport.java
index af388638dd1..e95d8237328 100644
--- a/maven/core/src/main/java/org/apache/karaf/maven/command/MavenConfigurationSupport.java
+++ b/maven/core/src/main/java/org/apache/karaf/maven/command/MavenConfigurationSupport.java
@@ -593,7 +593,7 @@ protected File nextSequenceFile(File dataDir, Pattern pattern, String fileNameFo
             List<String> names = new ArrayList<>(Arrays.stream(files).map(File::getName)
                     .collect(TreeSet<String>::new, TreeSet::add, TreeSet::addAll));
 
-            names.add(String.format(fileNameFormat, new Date().getTime()));
+            names.add(String.format(fileNameFormat, System.currentTimeMillis()));
 
             while (names.size() > MAX_SEQUENCE_SIZE) {
                 String name = names.remove(0);
