diff --git a/assembly/src/release/examples/openwire/ecommerce/src/Retailer.java b/assembly/src/release/examples/openwire/ecommerce/src/Retailer.java
index 01ecfef2832..5b764597dab 100644
--- a/assembly/src/release/examples/openwire/ecommerce/src/Retailer.java
+++ b/assembly/src/release/examples/openwire/ecommerce/src/Retailer.java
@@ -24,6 +24,7 @@
 import javax.jms.Session;
 import javax.jms.TemporaryQueue;
 
+import java.util.Random;
 /**
  * The Retailer orders computers from the Vendor by sending a message via
  * the VendorOrderQueue. It then syncronously receives the reponse message
@@ -58,7 +59,8 @@ public void run() {
 			for (int i = 0; i < 5; i++) {
 				MapMessage message = session.createMapMessage();
 				message.setString("Item", "Computer(s)");
-				int quantity = (int)(Math.random() * 4) + 1;
+				Random rand = new Random();
+				int quantity = (int)(rand.nextDouble() * 4) + 1;
 				message.setInt("Quantity", quantity);
 				message.setJMSReplyTo(retailerConfirmQueue);
 				producer.send(message);
