diff --git a/helix-core/src/main/java/org/apache/helix/tools/commandtools/ZkGrep.java b/helix-core/src/main/java/org/apache/helix/tools/commandtools/ZkGrep.java
index b42bdd278e..bb8b6701bf 100644
--- a/helix-core/src/main/java/org/apache/helix/tools/commandtools/ZkGrep.java
+++ b/helix-core/src/main/java/org/apache/helix/tools/commandtools/ZkGrep.java
@@ -581,7 +581,7 @@ public static void processCommandLineArgs(String[] cliArgs) {
     File zkParsedDir = new File(String.format("%s/zklog-parsed", System.getProperty("user.home")));
     if (!zkParsedDir.exists()) {
       LOG.info("creating zklog-parsed dir: " + zkParsedDir.getAbsolutePath());
-      zkParsedDir.mkdir();
+      zkParsedDir.mkdirs();
     }
 
     if (cmd.hasOption(between)) {
