diff --git a/core/src/main/java/com/taobao/arthas/core/command/monitor200/DashboardCommand.java b/core/src/main/java/com/taobao/arthas/core/command/monitor200/DashboardCommand.java
index 1ef53f95..571c3b05 100644
--- a/core/src/main/java/com/taobao/arthas/core/command/monitor200/DashboardCommand.java
+++ b/core/src/main/java/com/taobao/arthas/core/command/monitor200/DashboardCommand.java
@@ -321,7 +321,11 @@ public class DashboardCommand extends AnnotatedCommand {
                 addRuntimeInfo(dashboardModel);
 
                 //tomcat
-                addTomcatInfo(dashboardModel);
+                try {
+                    addTomcatInfo(dashboardModel);
+                } catch (Throwable e) {
+                    logger.error("try to read tomcat info error", e);
+                }
 
                 process.appendResult(dashboardModel);
 


--------
<0b20e4026c8612f9f1542415790a64f0a80f0180,d87861eb359aeedcafcd0c6affaee2f2e20184c7>
 | fix dashboard command may read tomcat info error. #1678