diff --git a/repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasTypeDefGraphStoreV2.java b/repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasTypeDefGraphStoreV2.java
index f360eef7f7..a5ccfb5b20 100644
--- a/repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasTypeDefGraphStoreV2.java
+++ b/repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasTypeDefGraphStoreV2.java
@@ -525,12 +525,11 @@ private void updateVertexProperty(AtlasVertex vertex, String propertyName, Date
      * increment the version value for this vertex
      */
     private void markVertexUpdated(AtlasVertex vertex) {
-        Date   now         = new Date();
         Number currVersion = vertex.getProperty(Constants.VERSION_PROPERTY_KEY, Number.class);
         long   newVersion  = currVersion == null ? 1 : (currVersion.longValue() + 1);
 
         vertex.setProperty(Constants.MODIFIED_BY_KEY, getCurrentUser());
-        vertex.setProperty(Constants.MODIFICATION_TIMESTAMP_PROPERTY_KEY, now.getTime());
+        vertex.setProperty(Constants.MODIFICATION_TIMESTAMP_PROPERTY_KEY, System.currentTimeMillis());
         vertex.setProperty(Constants.VERSION_PROPERTY_KEY, newVersion);
     }
 
