diff --git a/src/main/java/de/knightsoftnet/validators/shared/util/PhoneNumberUtil.java b/src/main/java/de/knightsoftnet/validators/shared/util/PhoneNumberUtil.java
index 3c98811..13e8d2f 100644
--- a/src/main/java/de/knightsoftnet/validators/shared/util/PhoneNumberUtil.java
+++ b/src/main/java/de/knightsoftnet/validators/shared/util/PhoneNumberUtil.java
@@ -221 +221 @@ public class PhoneNumberUtil {
-        if (phoneNumberWork.charAt(0) == '-') {
+        if (!phoneNumberWork.isEmpty() && phoneNumberWork.charAt(0) == '-') {
